### PR TITLE
Run the Kube Test Network on Rancher / k3s

### LINKED ISSUE #728: Kube test network : run on Rancher Desktop / k3s
KIND is good for local development, but it has some rough edges and can be improved.  As an alternate runtime for local development, use the [Rancher Desktop](https://rancherdesktop.io) and k3s as an alternate target for deployment on Kubernetes.

This is extremely useful in some environments, for example on Mac OSX without a Docker Desktop.

In addition, the k3s has visibility to the host's Docker image storage, so there is no longer a need to use the insecure Docker registry at localhost:5000 or load development images into the KIND control plane during local edit/test/build cycles.

Update the `./network` scripts such that it can support an alternate Kubernetes runtime.   Some issues that will need to be addressed in the scripts include: 

- Set STORAGE_CLASS=`local-path` when running on Rancher (was `standard` on KIND) 

- Update the Nginx ingress definitions to 1.1.2 and set `--enable-ssl-passthrough` on the controller CLI args: 
```
kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/controller-v1.1.2/deploy/static/provider/cloud/deploy.yaml
```
(Add these to manifests/namespace-install, or read from the base URL and kustomize the passthrough flag) 

- Update the README with a `RANCHER.md` guide 

- Add conditional branching for `TEST_NETWORK_CLUSTER_RUNTIME=rancher` to bypass any KIND-specific routines in the network scripts.

### PR DESCRIPTION
**Pull request by jkneubuh**
This PR enables users of the Kube Test Network to set up a Fabric network based on k3s / [Rancher Desktop](https://rancherdesktop.io)

- Fixes #728 
- Upgrades Nginx ingress to 1.1.2 
- Aligns both KIND and Rancher runtimes to use the local-path provisioner

Signed-off-by: Josh Kneubuhl <jkneubuh@us.ibm.com>

### CODE REVIEW

**Review by jt-nti — CHANGES_REQUESTED** [review]
Doesn't work on my machine :(

- scripts/prereqs.sh is checking for kind even when TEST_NETWORK_CLUSTER_RUNTIME is set to k3s
- there is no `envsubst` command on OSX so get errors, e.g. `scripts/fabric_config.sh: line 35: envsubst: command not found` (think this probably just needs to be in prereqs.sh as well)

File: test-network-k8s/docs/KUBERNETES.md [hunk 1]
```diff
@@ -69,6 +69,36 @@ or:
...
+   4. Increase CPU allocation to 8 CPU 
+
+2. Reset Kubernetes 
+
+3. Initialize the Nginx ingress and cert-manager: 
+
+```shell
+export TEST_NETWORK_CLUSTER_RUNTIME=k3s
+
+./network cluster-init
```
Comment (jt-nti) [inline test-network-k8s/docs/KUBERNETES.md:91]: Just had another thought- is there an equivalent to `unkind` for this?

Comment (jkneubuh) [reply test-network-k8s/docs/KUBERNETES.md:91]: Rancher desktop maintains a 1:1 "rancher" : "kubernetes is running" strategy.   So there's no real way to delete the rancher kube without going through the Rancher GUI and hitting the "reset Kubernetes" button.

Not to mention ... I couldn't find a verb that served as the inverse of "to rancher."  E.g.  `./network unrancher` doesn't have the same zing as `unkind`  -- I did try to find a good word for this, but abandoned the effort. 

I did make some changes to the teardown / `network down` script, to better scrub the kubernetes cluster of test network artifacts.

Comment (jt-nti) [reply test-network-k8s/docs/KUBERNETES.md:91]: Just wondering what undoes what `cluster-init` did really. I'm assuming that `network down` won't undo this?

```
✅ - Launching k3s ingress controller ...
✅ - Launching cert-manager ...
✅ - Waiting for cert-manager ...
✅ - Waiting for ingress controller ...
```

Maybe rename `network unkind` to `network clean`? Actually a generic `network prereqs` or some such which does the right thing depending on `TEST_NETWORK_CLUSTER_RUNTIME` might be better than having to pick between running `network kind` or `network cluster-init`?

Not worth holding this PR up for though.

**Comment by jkneubuh** [comment]
There is some crossover in this PR with Fabric [Issue #3372](https://github.com/hyperledger/fabric/issues/3372).

The latest commit on this PR works around a slow DNS resolution issue (5s / timeout) by increasing the timeout for connections to the orderer with the peer `--connTimeout` flag.

We can, should, and will fix the DNS resolver issues on OSX.  Just - not in this ticket / PR.

**Comment by jkneubuh** [comment]
Thanks @denyeart for the test and validation / troubleshooting.  The latest commit introduces:

- 10s delay for the peer delivery client connection timeout in the core configs. 
- Highlights the prerequisite for 8 CPU / 8GRAM in the docker / rancher runtimes.

**Comment by mbwhite** [comment]
Hello @jkneubuh.... 

- the different start comments for the cluster - can these be combined? Something along these lines... (with one of them being a default)
```
export TEST_NETWORK_CLUSTER_RUNTIME=k3s
./network cluster-init 

export TEST_NETWORK_CLUSTER_RUNTIME=kind
./network cluster-init 
```

- probably error handling messages are the weakest point of an otherwise a winning script.

**Comment by jkneubuh** [comment]
OK.  This PR is (over) cooked and done.

- Fixes #506 (pretty prints an error message AND an emoji on error) 
- Fixes #728 (runs with both containerd/nerdctl and mobyd/docker) 
- Fixes #550 (or verified that this is indeed addressed)
- Includes a good workaround for [Fabric #3372](https://github.com/hyperledger/fabric/issues/3372) (slow DNS on OSX)
- Ironically - test flakes in the CI pipeline were due to collisions (unrelated) to a Docker Compose bug.

_Goodbye, Docker!_

**Pull request merged**